teuthology/exceptions.py

```diff
@@ -50,21 +54,30 @@ class CommandFailedError(Exception):
     """
     Exception thrown on command failure
     """
-    def __init__(self, command, exitstatus, node=None, label=None):
-        self.command = command
+    def __init__(self, cmd, exitstatus, node=None, label=None):
+        self.cmd = cmd
         self.exitstatus = exitstatus
         self.node = node
         self.label = label
 
     def __str__(self):
+        if isinstance(self.cmd, (list, tuple)):
+            self.cmd = [str(Raw) if isinstance(x, Raw) else x \
+                            for x in self.cmd]
+            self.cmd = ' '.join(self.cmd)
+        elif isinstance(self.cmd, str):
+            self.cmd = sub(r'Raw\([\'"](.*)[\'"]\)', r'\1', self.cmd)
```

Comment:
Why this branch is needed?

Reply:
Transforms str `'param1 param2 Raw('&&") param3'` to str `'param1 param2 && param3'`.

Reply:
why this is needed to transform I do not understand

Reply:
or in which situation this is happening...

Reply:
Same reason for which i am converting `list` to `str`.

Reply:
well... I don't understand why we do not use quote method here from the helper?

Reply:
How would that do the job?

```
$ python
Python 3.8.9 (default, Apr  6 2021, 00:00:00) 
[GCC 10.2.1 20201125 (Red Hat 10.2.1-9)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from teuthology.orchestra.run_helper import Raw, quote
>>> a = quote('a b Raw("&&") c')
>>> a
'a b Raw("&&") c'
>>> 
```

Reply:
Yep, but how is it possible to have the Raw in the string? From my point of view if you have Raw in the str it means that you've made a typo and this command is not going to be run successfully.